File: ios/App/Sources/Core/Domain/NavCaddy/Module.swift
import Foundation

/// Represents the main modules in the CaddyPro app.
///
/// Used for routing intents to the correct module.
enum Module: String, Codable, CaseIterable, Hashable {
    case caddy = "CADDY"
    case coach = "COACH"
    case recovery = "RECOVERY"
    case settings = "SETTINGS"
}

File: android/app/src/main/java/com/example/app/domain/navcaddy/models/Module.kt
package caddypro.domain.navcaddy.models

/**
 * Enum representing the primary app modules for intent routing.
 *
 * Each module represents a major functional area of the app that can be
 * targeted by user intents.
 *
 * Spec reference: navcaddy-engine.md R3
 */
enum class Module {
    /** Live caddy mode - shot recommendations, yardage, strategy */
    CADDY,

    /** Coaching and drill suggestions */
    COACH,

    /** Recovery tracking and wellness insights */
    RECOVERY,

    /** App settings and preferences */
    SETTINGS
}
